packages/identity/src/identity/der.ts

```diff
@@ -0,0 +1,113 @@
+const bufEquals = (b1: ArrayBuffer, b2: ArrayBuffer): boolean => {
+  if (b1.byteLength !== b2.byteLength) return false;
+  const u1 = new Uint8Array(b1);
+  const u2 = new Uint8Array(b2);
+  for (let i = 0; i < u1.length; i++) {
+    if (u1[i] !== u2[i]) return false;
+  }
+  return true;
+};
+
+const encodeLenBytes = (len: number): number => {
+  if (len <= 0x7f) {
+    return 1;
+  } else if (len <= 0xff) {
+    return 2;
+  } else if (len <= 0xffff) {
+    return 3;
+  } else if (len <= 0xffffff) {
+    return 4;
+  } else {
+    throw new Error('Length too long (> 4 bytes)');
+  }
+};
+
+const encodeLen = (buf: Uint8Array, offset: number, len: number): number => {
+  if (len <= 0x7f) {
+    buf[offset] = len;
+    return 1;
+  } else if (len <= 0xff) {
+    buf[offset] = 0x81;
+    buf[offset + 1] = len;
+    return 2;
+  } else if (len <= 0xffff) {
+    buf[offset] = 0x82;
+    buf[offset + 1] = len >> 8;
+    buf[offset + 2] = len;
+    return 3;
+  } else if (len <= 0xffffff) {
+    buf[offset] = 0x83;
+    buf[offset + 1] = len >> 16;
+    buf[offset + 2] = len >> 8;
+    buf[offset + 3] = len;
+    return 4;
+  } else {
+    throw new Error('Length too long (> 4 bytes)');
+  }
+};
+
+const decodeLenBytes = (buf: Uint8Array, offset: number): number => {
+  if (buf[offset] < 0x80) return 1;
+  if (buf[offset] === 0x80) throw new Error('Invalid length 0');
+  if (buf[offset] === 0x81) return 2;
+  if (buf[offset] === 0x82) return 3;
+  if (buf[offset] === 0x83) return 4;
+  throw new Error('Length too long (> 4 bytes)');
+};
+
+export const DER_COSE_OID = Uint8Array.from([
```

Comment:
Your code should do the correct thing, so this is just a note re: naming. The OID is technically only the part starting with `0x06` – the `0x03 0x0c` is another `SEQUENCE` wrapping the OID.

The reason for the `SEQUENCE` is that some OIDs come with an additional parameter; the parameter just happens to be empty for both Ed25519 and DER-wrapper-COSE. For other schemes, that is different. For ECDSA, the OID defining the scheme requires an additional OID defining the elliptic curve used.

What I want to say: you are encoding `SEQUENCE(OID)` here, and for ECDSA what you would see is `SEQUENCE(OID, OID)` – so the `SEQUENCE` is not part of the `OID`. Makes sense?

Reply:
Right. I figured the extra nuance doesn't mean much outside of this module, as you'd never tease the encoded OID apart for our use-cases so I didn't include it in the name/api. Think I should add your comment as an explanation into the source anyway?